go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -203,13 +215,20 @@ func (ncc *networkClusterController) syncNodeClusterSubnet(node *corev1.Node) er
 		klog.Warningf("Failed to get node %s host subnets annotations for network %s : %v", node.Name, ncc.networkName, err)
 	}
 
+	networkID, err := util.ParseNetworkIDAnnotation(node, ncc.networkName)
```

Comment:
Isn't it a bit weird that this is annotated in the node? I agree that we don't have a great place to store this information. It's more natural place would be the NAD but that requires an effort hrrrmm

Reply:
Actually we should have a Network CRD but the reluctance we have to introduce CRDs saddens me a bit :P

Reply:
A user would need permission to annotate a node so I think it is relatively safe and convenient.

Reply:
Yes but we are replicating the same information in all the nodes opening the door to the possibility that one node has different information than other node for whatever reason.